src/main/java/com/mixpanel/android/mpmetrics/Tweaks.java

```diff
@@ -109,12 +145,39 @@ public void set(Map<String, Object> tweakUpdates) {
         }
     }
 
-    private synchronized void set(String tweakName, Object value, boolean isDefault) {
-        mTweaks.put(tweakName, value);
-
-        if (!isDefault && mBindings.containsKey(tweakName)) {
-            for(TweakChangeCallback changeCallback : mBindings.get(tweakName)) {
-                runCallback(changeCallback, value);
+    public void registerForTweaks(Object registrant) {
+        synchronized (sRegistrars) {
+            Class klass = registrant.getClass();
+            while (klass != Object.class) {
+                if (! sRegistrars.containsKey(klass)) {
+                    final ClassLoader loader = klass.getClassLoader();
+                    final Package registrantPackage = klass.getPackage();
+
+                    try {
+                        final Class found = loader.loadClass(registrantPackage.getName() + "." + mTweakClassName);
+                        final Field instanceField = found.getField("TWEAK_REGISTRAR");
+                        final TweakRegistrar registrar = (TweakRegistrar) instanceField.get(null);
+                        sRegistrars.put(klass, registrar);
+                    } catch (ClassNotFoundException e) {
+                        ; // Ok, no such class.
```

Comment:
This will happen more often than not, and it isn't an error condition- we just don't have a `ClassLoader.classExists` method to call.